File: example/fs/Samples/Custom.fs
namespace Samples

open AUTD3Sharp
open AUTD3Sharp.Gain
open AUTD3Sharp.Modulation
open type AUTD3Sharp.Units

module CustomTest =
    let Test<'T> (autd : Controller<'T>) = 
        (Silencer.Default()) |> autd.Send;

        let m = new Sine(150u * Hz);
        let g = new Custom(fun dev -> new System.Func<Transducer, Drive>(fun tr -> match (dev.Idx, tr.Idx) with
                                                                                    | (0, 0) -> new Drive(new Phase(byte(0)), EmitIntensity.Max)
                                                                                    | (0, 248) -> new Drive(new Phase(byte(0)), EmitIntensity.Max)
                                                                                    | _ ->  Drive.Null));
        (m, g) |> autd.Send;

File: example/Samples/Custom.cs
using AUTD3Sharp;
using AUTD3Sharp.Modulation;
using static AUTD3Sharp.Units;

namespace Samples;

internal static class CustomGain
{
    public static void Test(Controller autd)
    {
        var config = new Silencer();
        autd.Send(config);

        var m = new Sine(freq: 150 * Hz, option: new SineOption());
        var g = new AUTD3Sharp.Gain.Custom(
            (dev) => (tr) => (dev.Idx(), tr.Idx()) switch
            {
                (0, 0) => new Drive
                {
                    Phase = Phase.Zero,
                    Intensity = EmitIntensity.Max
                },
                (0, 248) => new Drive
                {
                    Phase = Phase.Zero,
                    Intensity = EmitIntensity.Max
                },
                _ => Drive.Null
            }
        );
        autd.Send((m, g));
    }
}
